source/fundamentals/gridfs.txt

```diff
@@ -4,7 +4,369 @@ GridFS
+a file ID, file name, and other file metadata. The following diagram shows
+how GridFS splits the uploaded files:
+
+.. figure:: /includes/figures/GridFS-upload.png
+   :alt: A diagram that shows how GridFS uploads a file to a bucket
+
+When retrieving files, GridFS fetches the metadata from the ``files``
+collection in the specified bucket, then uses that information to reconstruct
+the file from documents in the ``chunks`` collection. You can read the file
+into memory or output it to a stream.
+
+Use GridFS
+----------
+
+To learn about GridFS operations and how to perform them, navigate to the
+following sections:
+
+- :ref:`<golang-create-bucket>`
+- :ref:`<golang-upload-files>`
+- :ref:`<golang-retrieve-info>`
+- :ref:`<golang-download-files>`
+- :ref:`<golang-rename-files>`
+- :ref:`<golang-delete-files>`
+- :ref:`<golang-delete-bucket>`
+
+.. _golang-create-bucket:
+
+Create a GridFS Bucket
+~~~~~~~~~~~~~~~~~~~~~~
+
+To store or retrieve files from GridFS, create a bucket or get a reference to
+an existing bucket on a MongoDB database. To create a ``GridFSBucket`` instance,
+call the ``NewBucket()`` method with a database parameter:
+
+.. code-block:: go
+
+   db := client.Database("myDB")
+   bucket, err := gridfs.NewBucket(db)
+   if err != nil {
+      panic(err)
+   }
+
+.. note::
+
+   If a GridFS bucket already exists, the ``NewBucket()`` method returns a
+   reference to the bucket rather than instantiating a new one.
+
+By default, the new bucket is named ``fs``. To instantiate a bucket with a
+custom name, call the ``SetName()`` method on a ``BucketOptions`` instance as 
+follows:
+
+.. code-block:: go
+
+   db := client.Database("myDB")
+   opts := options.GridFSBucket().SetName("custom name")
+   bucket, err := gridfs.NewBucket(db, opts)
+   
+   if err != nil {
+      panic(err)
+   }
+
+.. _golang-upload-files:
+
+Upload Files
+~~~~~~~~~~~~
+
+You can upload a file into a GridFS bucket in one of the following ways:
+
+- Use the ``UploadFromStream()`` method, which reads from an input stream.
+- Use the ``OpenUploadStream()`` method, which writes to an output stream.
+
+For either upload process, you can specify configuration information on an instance
+of ``UploadOptions``. For a full list of ``UploadOptions`` fields, visit the 
+`API documentation <{+api+}/mongo/options#UploadOptions>`__.
+
+Upload with an Input Stream
+```````````````````````````
+
+To upload a file with an input stream, use the ``UploadFromStream()`` method
+with the following parameters:
+
+- Your file name
+- An ``io.Reader``, with your opened file as a parameter
+- An optional ``opts`` parameter to modify the behavior of ``UploadFromStream()``
+
+The following code example reads from a file called ``file.txt`` and uploads the
+content to a GridFS bucket. It uses an ``opts`` parameter to set file metadata:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input::
+      :language: go
+
+      fileContent, err := os.Open("path/to/file.txt")
```

Comment:
```suggestion
      file, err := os.Open("path/to/file.txt")
      if err != nil {
         panic (err);
      }
```

Suggest renaming `fileContent` to  `file`. `os.Open` returns `*File`. The contents are not yet read.

Reply:
I renamed `fileContent` to `file`.